lib/config/config-initializer.js

```diff
@@ -239,43 +242,52 @@ function configureRules(answers, config) {
  * @returns {Object} config object
  */
 function processAnswers(answers) {
-    let config = { rules: {}, env: {}, parserOptions: {} };
+    let config = {
+        rules: {},
+        env: {},
+        parserOptions: {}
+    };
 
-    config.parserOptions.ecmaVersion = answers.ecmaVersion;
-    if (answers.ecmaVersion >= 2015) {
-        if (answers.modules) {
-            config.parserOptions.sourceType = "module";
-        }
-        config.env.es6 = true;
-    }
+    // set the latest ECMAScript version
+    config.parserOptions.ecmaVersion = DEFAULT_ECMA_VERSION;
+    config.env.es6 = true;
 
-    if (answers.commonjs) {
+    // set the module type
+    if (answers.moduleType === "esm") {
+        config.parserOptions.sourceType = "module";
+    } else if (answers.moduleType === "commonjs") {
         config.env.commonjs = true;
     }
+
+    // add in browser and node environments if necessary
     answers.env.forEach(env => {
         config.env[env] = true;
     });
-    if (answers.jsx) {
-        config.parserOptions = config.parserOptions || {};
-        config.parserOptions.ecmaFeatures = config.parserOptions.ecmaFeatures || {};
-        config.parserOptions.ecmaFeatures.jsx = true;
-        if (answers.react) {
-            config.plugins = ["react"];
-            config.parserOptions.ecmaVersion = 2018;
-        }
-    }
 
-    if (answers.source === "prompt") {
-        config.extends = "eslint:recommended";
-        config.rules.indent = ["error", answers.indent];
-        config.rules.quotes = ["error", answers.quotes];
-        config.rules["linebreak-style"] = ["error", answers.linebreak];
-        config.rules.semi = ["error", answers.semi ? "always" : "never"];
+    // add in library information
+    if (answers.framework === "react") {
+        config.parserOptions.ecmaFeatures = {
+            jsx: true
+        };
+        config.plugins = ["react"];
+    } else if (answers.framework === "vue") {
+        config.plugins = ["vue"];
```

Comment:
There is a special parser for Vue.js. (Also don't forget to update the function `getModulesList`.)
```suggestion
        config.parser = "vue-eslint-parser";
        config.plugins = ["vue"];
```

Reply:
Is the Vue parser required for working Vue? Or is it optional?

Reply:
It's required.

Reply:
Okay, thank you. I will add that in.

Reply:
Actually, `eslint-plugin-vue` contains `vue-eslint-parser` as a dependency and it requires some additional configuration to work properly. The plugin provides `plugin:vue/base` preset for this.

> https://github.com/vuejs/eslint-plugin-vue/blob/f46c75b391422a651711fe67aac60963faf39406/lib/configs/base.js

- `parser: require.resolve('vue-eslint-parser'),` supports parsing of `*.vue` files.
- `'vue/comment-directive': 'error',` supports `<!-- eslint-disable -->` comment directives in Vue.js template.
- `parserOptions.ecmaFeatures.jsx: true` and `'vue/jsx-uses-vars': 'error'` supports JSX syntax.

Then, they provides `plugin:vue/essential` preset to catch problems in Vue.js files as smilar to `eslint:recommended`.

> https://github.com/vuejs/eslint-plugin-vue/blob/f46c75b391422a651711fe67aac60963faf39406/lib/configs/essential.js

Those rules catch only semantic errors.

Therefore, I'd like to recommend to use `extends: "plugin:vue/essential"` instead.

```suggestion
        config.extends = ["plugin:vue/essential"];
```

But if you don't want to contain rules which catch semantic errors, `plugin:vue/base` (configuration to parse `*.vue` files properly) is a good option.

Reply:
Sounds good, I'll make these changes.

Reply:
Thank you! LGTM about Vue's setting.